service/history/shard/context_impl.go

```diff
@@ -1291,11 +1291,13 @@ func (s *ContextImpl) start() {
 }
 
 func (s *ContextImpl) Unload() {
-	s.stop()
+	s.wLock()
+	defer s.wUnlock()
+	s.transitionLocked(contextRequestStop{})
 }
 
-// stop should only be called by the controller.
-func (s *ContextImpl) stop() {
+// finishStop should only be called by the controller.
+func (s *ContextImpl) finishStop() {
```

Comment:
I usually name the method start with _ if i don't want others to call it, so it would be _stop(). That way, when people make calls to it, they will look closer.
But i'm not sure if that is best practice.

Reply:
That feels like a Python idiom to me. There aren't any methods or functions named with initial underscore in this repo, except for some proto-generated code. I think this is fine (after fixing the test).